### ISSUE TITLE
Support Terraform Data to pull data from the Port Catalog

### ISSUE THREAD

**Issue by dexter0195** [port-labs/terraform-provider-port-labs#126]
Hello !
It would be awesome if we could query Port and use that data in our terraform, right now we can still do it with some terraform variables or by using the http data provider but it would be nice if that would be supported natively. 

I post here the workaround I am using in case someone is interested in the same feature : 
```hcl
data "http" "my_datatacenters" {
  url = "https://api.getport.io/v1/blueprints/terraform_output/entities/my_datatacenters"
  request_headers = {
    Content-Type = "application/json; charset=utf8"
    Authorization: "Bearer XXXX"
  }
  method = "GET"
}

output "my_datatacenters" {
  value = jsondecode(data.http.my_datatacenters.response_body).entity.properties.raw_json_output.pprd.eus1.0
}
```

Thank you 😃 

I would be interested in contributing in case you think this feature is interesting

**Comment by Tankilevitch** [comment]
Hey @dexter0195 👋 
That sounds like a great idea!!! and definitely a feature we are planning to implement in the future.
It will be amazing if you would contribute that capability. 
Let me know if you would also like to schedule a meeting to go over the provider 😄

**Comment by Tankilevitch** [comment]
Hey @dexter0195 👋 

We just released support to use `search` as datasource in the provider.  #153 

It was released in [2.0.6](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/data-sources/port_search) 😄
